Frontend/src/pages/registration_administration/components/RegistrationAdministrationList.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react'
+import deleteRegistration from '../../../api/registration/delete/delete_registration'
+import { getAllRegistrations } from '../../../api/registration/get/get_registrations'
+import updateRegistration from '../../../api/registration/patch/update_registration'
+import styles from './list.module.scss'
+import StatusDropdown from './StatusDropdown'
+
+function RegistrationRow({
+  competitorId,
+  eventIDs,
+  serverStatus,
+  competitionID,
+  setRegistrationList,
+  registrationList,
+}) {
+  const [status, setStatus] = useState(serverStatus)
+
+  return (
+    <tr>
+      <td>{competitorId}</td>
+      <td>{eventIDs.join(',')}</td>
+      <td>
+        <StatusDropdown status={status} setStatus={setStatus} />
+      </td>
+      <td>
+        <button
+          onClick={(_) => {
+            updateRegistration(competitorId, competitionID, status)
+          }}
+        >
+          {' '}
+          Apply
+        </button>
+      </td>
+      <td>
+        <button
+          onClick={(_) => {
+            deleteRegistration(competitorId, competitionID)
+            setRegistrationList(
+              registrationList.filter((r) => r.competitor_id !== competitorId)
+            )
+          }}
+        >
+          Delete
+        </button>
+      </td>
+    </tr>
+  )
+}
+
+export default function RegistrationAdministrationList() {
```

Comment:
Keeping the default function at the top is good; function dependencies pointing downward in a file is nice.